src/doc/src/reference/features.md

```diff
@@ -1,134 +1,456 @@
-enabled using command-line flags:
+```toml
+[dependencies]
+flate2 = { version = "1.0.3", default-features = false, features = ["zlib"] }
+```
+
+> **Note**: This may not ensure the default features are disabled. If another
+> dependency includes `flate2` without specifying `default-features = false`,
+> then the default features will be enabled. See [feature
+> unification](#feature-unification) below for more details.
+
+Features of dependencies can also be enabled in the `[features]` table. The
+syntax is `"package-name/feature-name"`. For example:
+
+```toml
+[dependencies]
+jpeg-decoder = { version = "0.1.20", default-features = false }
 
-```console
-$ cargo build --release --features "shumway pdf"
+[features]
+# Enables parallel processing support by enabling the "rayon" feature of jpeg-decoder.
+parallel = ["jpeg-decoder/rayon"]
 ```
 
-Default features could be excluded using `--no-default-features`.
+> **Note**: The `"package-name/feature-name"` syntax will also enable
+> `package-name` if it is an optional dependency. Experimental support for
+> disabling that behavior is available on the nightly channel via [weak
+> dependency features](unstable.md#weak-dependency-features).
 
-### Usage in packages
+### Command-line feature options
 
-In most cases, the concept of *optional dependency* in a library is best
-expressed as a separate package that the top-level application depends on.
+The following command-line flags can be used to control which features are
+enabled:
 
-However, high-level packages, like Iron or Piston, may want the ability to
-curate a number of packages for easy installation. The current Cargo system
-allows them to curate a number of mandatory dependencies into a single package
-for easy installation.
+* `--features` _FEATURES_: Enables the listed features. Multiple features may
+  be separated with commas or spaces. If using spaces, be sure to use quotes
+  around all the features if running Cargo from a shell (such as `--features
+  "foo bar"`). If building multiple packages in a [workspace], the
+  `package-name/feature-name` syntax can be used to specify features for
+  specific workspace members.
 
-In some cases, packages may want to provide additional curation for optional
-dependencies:
+  > **Note**: With the `resolver = "1"` option [described
+  > below](#feature-resolver-version-2), the features chosen are for the
+  > package in the current directory, not the packages selected with the `-p`
+  > flags. With the `resolver = "2"`, it will instead enable the listed
+  > features for the packages listed in the `-p` flags.
+  >
+  > For example, when `resolver = "2"` is enabled:
+  >
+  >     cargo build -p member1 -p member2 --features foo,bar
+  >
+  > In this situation, features "foo" and "bar" are enabled on the given
+  > members only if the member defines that feature. It is an error if none of
+  > the selected packages defines a given feature.
```

Comment:
I'm a bit on the fence about the inline notes here about resolver=1/2. I find them useful personally but as the ecosystem shifts more and more to =2 then these notes will largely become clutter. Perhaps this chapter could be written from the perspective of a year from now with notes about resolver=1 at the end?

Reply:
Sure! That's something I struggle with in the Rust reference, too, trying to describe multiple editions at the same time. I removed the notes, since they were essentially just duplicating the same information, and aren't that important. I think in the future when "2" is the default, it can be tweaked some more.